proposals/0233-codepoint-and-character-literals.md

```diff
@@ -0,0 +1,234 @@
+
+Single-quoted literals will be inferred to be integer types in cases where a `Character` or `Unicode.Scalar` overload does not exist, but an integer overload does. This can lead to strange spellings such as `'1' + 1' == 98`. However, we forsee problems arising from this to be quite rare, as the type system will almost always catch such mistakes, and very few users are likely to express a `String` with two literals instead of the much more obvious `"11"`. 
+
+Use of single quotes for character/scalar literals is *heavily* precedented in other languages, including C, Objective-C, C++, Java, and Rust, although different languages have slightly differing ideas about what a “character” is.  We choose to use the single quote syntax specifically because it reinforces the notion that strings and character values are different: the former is a sequence, the later is a scalar (and "integer-like").  Character types also don't support string literal interpolation, which is another reason to move away from double quotes.
+
+### Single quotes in Swift, a historical perspective
+
+In Swift 1.0, we wanted to reserve single quotes for some yet-to-be determined syntactical purpose. However, today, pretty much all of the things that we once thought we might want to use single quotes for have already found homes in other parts of the Swift syntactical space.  For example, syntax for [multi-line string literals](https://github.com/apple/swift-evolution/blob/master/proposals/0168-multi-line-string-literals.md) uses triple quotes (`"""`), and string interpolation syntax uses standard double quote syntax. With the passage of [SE-0200](https://github.com/apple/swift-evolution/blob/master/proposals/0200-raw-string-escaping.md), raw-mode string literals settled into the `#""#` syntax. In current discussions around [regex literals](https://forums.swift.org/t/string-update/7398/6), most people seem to prefer slashes (`/`).
+
+At this point, it is clear that the early syntactic conservatism was unwarranted.  We do not forsee another use for this syntax, and given the strong precedent in other languages for characters, it is natural to use it.
+
+### Existing double quote initializers for characters
+
+We propose deprecating the double quote literal form for `Character` and `Unicode.Scalar` types and slowly migrating them out of Swift.
+
+```swift
+let c2 = 'f'               // preferred
+let c1: Character = "f"   // deprecated
+```
+
+## Detailed Design 
+
+The only standard library change will be to add `{UInt8, Int8, ..., Int}` to the list of allowed `Self.UnicodeScalarLiteralType` types. (This entails conforming the integer types to `_ExpressibleByBuiltinUnicodeScalarLiteral`.) The ASCII range checking will be performed at compile-time in the typechecker, in essentially the same way that overflow checking for `ExpressibleByIntegerLiteral.IntegerLiteralType` types works today.
```

Comment:
"The only additional standard library change..."? Adding `ExpressibleByUnicodeScalarLiteral` conformance to the integer types is the other change.

Reply:
done